packages/client/wallets/smart-wallet/src/api/CrossmintWalletService.ts

```diff
@@ -37,22 +37,22 @@ export class CrossmintWalletService extends BaseCrossmintService {
         return parseBigintAPIResponse(result);
     }
 
-    async getSmartWalletConfig(
-        user: UserParams,
-        chain: SmartWalletChain
-    ): Promise<{
-        kernelVersion: string;
-        entryPointVersion: string;
-        userId: string;
-        signers: { signerData: SignerData }[];
-        smartContractWalletAddress?: string;
-    }> {
-        return this.fetchCrossmintAPI(
+    async getSmartWalletConfig(user: UserParams, chain: SmartWalletChain): Promise<SmartWalletConfig> {
+        const data: unknown = await this.fetchCrossmintAPI(
             `${API_VERSION}/sdk/smart-wallet/config?chain=${chain}`,
             { method: "GET" },
             "Error getting smart wallet version configuration. Please contact support",
             user.jwt
         );
+
+        const result = SmartWalletConfigSchema.safeParse(data);
+        if (result.success) {
+            return result.data;
+        }
+
+        throw new CrossmintServiceError(
```

Comment:
Wonder if it would make sense to make a utility function that calls safeParse and throws a CrossmintServiceError if it fails

Reply:
I think so, but let's wait until there's a couple more instances of this pattern to be sure